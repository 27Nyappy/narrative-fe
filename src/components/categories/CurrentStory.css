.CurrentStory {
  background-color: grey;
  display: flex;

  div {
    background-color: palegoldenrod;
    width: 50%;
  }
}