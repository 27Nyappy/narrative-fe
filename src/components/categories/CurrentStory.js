import React, { useState } from 'react';
import PropTypes from 'prop-types';
import DefaultViewModalEdit from '../../containers/default-view/DefaultView-Modal-Edit';

const CurrentStory = ({ title, synopsis }) => {

  const [show, setShow] = useState(false);

  const showModal = () => {
    setShow(true);
  };

  const hideModal = () => {
    setShow(false);
  };

  return (
    <section>
      <div>
        <h2>{title}</h2>
        <DefaultViewModalEdit show={show} hideModal={hideModal} />
        <button type='button' onClick={showModal}>Edit Story</button>
      </div>

      <div>
        <p>{synopsis}</p>
      </div>
    </section>
  );
};

CurrentStory.propTypes = {
  title: PropTypes.string,
  synopsis: PropTypes.string
};

export default CurrentStory;