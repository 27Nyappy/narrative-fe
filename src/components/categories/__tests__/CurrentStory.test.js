import React from 'react';
import { shallow } from 'enzyme';
import CurrentStory from '../CurrentStory';

describe('CurrentStory component', () => {
  it('renders a current story', () => {
    const wrapper = shallow(<CurrentStory title="current story" handleEditClick={() => true} synopsis="this is the current story" />);
    expect(wrapper).toMatchSnapshot();
  });
});