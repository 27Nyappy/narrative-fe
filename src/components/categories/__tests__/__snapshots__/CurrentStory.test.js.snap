// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentStory component renders a current story 1`] = `
<section>
  <div>
    <h2>
      current story
    </h2>
    <button
      onClick={[Function]}
    >
      Edit
    </button>
  </div>
  <div>
    <p>
      this is the current story
    </p>
  </div>
</section>
`;
